lib/src/thir_pattern.rs

```diff
@@ -100,7 +100,21 @@ pub(crate) fn compile_pattern(env: &Env, pat: &Pat) -> Rc<Pattern> {
         PatKind::Constant { value } => {
             if let rustc_middle::mir::Const::Ty(constant) = value {
                 let ty = constant.ty();
-
+                // Brutal way to handle the case of rustc_middle::ty::TyKind::Str
+                // Since the type would be erased when it comes down to thir level
```

Comment:
```suggestion
                // Since the type would be erased when it comes down to THIR level
                // TODO: have a translation that works for all strings
```
I guess that for strings that contain special characters such as `"` we would have a limitation.

Reply:
I believe that the trim should work fine. There would always have an extra `""` pair wrapped around the original string, and I trimmed them out. The rest of the content are being processed by `string_pieces_to_coq` which handles such case.